src/citrine/resources/ara_definition.py

```diff
@@ -161,3 +169,29 @@ def preview(self, defn: AraDefinition, preview_roots: List[LinkByUID]) -> dict:
             "rows": [x.as_dict() for x in preview_roots]
         }
         return self.session.post_resource(path, body)
+
+    def register(self, defn: AraDefinition) -> AraDefinition:
```

Comment:
@dekatzenel could you comment on the decision not to use `update` here?  Is it because of the lack of server-side update validation?

Reply:
Note: I don't have strong feelings here and can definitely switch it if requested. The reasoning I was using was based on:

1. Like you said, the validation requirements for an AraDefinition do not change between when it is registered vs updated, which means that there doesn't need to be a substantial UX difference between register & update
2. This prevents users from accidentally registering duplicate AraDefinitions when they modify an existing one, which is presumably a thing we want to do based on the design model (though I could also imagine wanting to facilitate leaving old ones alone)

When you said comment, you meant in the code, right? Commenting here first in case consensus is to switch to using update given my reasoning.

Reply:
Both here and in code :-P.  I agree that we want to do (2), but it could be by raising a ValueError if the defn passed into `register` has a `definition_id` assigned already.